api/envoy/config/route/v3/route_components.proto

```diff
@@ -1885,8 +1885,8 @@ message HeaderMatcher {
     //   "-1somestring"
     type.v3.Int64Range range_match = 6;
 
-    // If specified, header match will be performed based on whether the header is in the
-    // request.
+    // If specified as true, header match will be performed based on whether the header is in the
+    // request. If specified as false, header match will be performed based on whether the header is absent.
```

Comment:
I think this is technically a data plane behavioral change. That said, I don't see any sensible way that this field could have been configured with `false` and a meaningful match expected previously, so I don't think we need runtime protection.

Reply:
the doc is not correct, config `present_match: false` will not effected, shoud use 
```
present_match: true
invert_match:true
```

Reply:
@hanjm I tested it on the main branch, it works as expected. I pasted the detail at https://github.com/envoyproxy/envoy/issues/17345#issuecomment-1177455864

Reply:
sorry, my version(1.17) is too low. Thanks you reply.